jax/_src/plugin_utils.py

```diff
@@ -0,0 +1,66 @@
+# Copyright 2023 The JAX Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import pathlib
+import glob
+
+# Code to suggest plugins that should be installed.
+#
+# Plugin vendors are welcome to add code to this list, assuming there's a
+# lightweight way to determine if hardware is present without requiring
+# the relevant plugin be installed.
+
+_GOOGLE_PCI_VENDOR_ID = '0x1ae0'
+_TPU_PCI_DEVICE_IDS = [
+    # TPU v2, v3
+    '0x0027',
+    # TPU v4
+    '0x005e',
+    # TPU v5e
+    '0x0063',
+    # Testing only
+    '0x0056',
+    '0x0062',
+]
+
+_TPU_ENHANCED_BARRIER_SUPPORTED = [
+    # TPU v2, v3
+    '0x0027',
+    # TPU v4
+    '0x005e',
+]
+
+def _num_available_tpu_chips_device_id():
```

Comment:
nit: since this is meant to be used outside this file, it shouldn't have a leading underscore. It's still not a user-facing public jax function because we don't import it outside of the _src directory, just a public function from the perspective of other internal _src code.

Also let's called `num_available_tpu_chips_and_device_id`, which makes it a bit easier to read.